core/server/data/importer/importers/data/posts.js

```diff
@@ -160,6 +161,11 @@ class PostsImporter extends BaseImporter {
                     model.comment_id = model.id;
                 }
             }
+
+            // @TODO: test if html field is missing, test if mobiledoc field is missing (throw error?)
+            if (model.html && model.html.match(/^<div class="kg-card-markdown">/)) {
```

Comment:
Which versions will Ghost 2.0 allow imports from? If imports are allowed from versions < 1.25.0 then we may also have Koenig beta posts to deal with that will need the mobiledoc updating before re-rendering 🤔

Reply:
This probably needs a little discussion 🤔What we said so far is: we will force people to upgrade their 1.0 blog to the latest version in the CLI before they can jump to 2.0. But we haven't talked about the importer.

Reply:
Hmm, I don't think it would be too hard to support all versions. It would be quite annoying for users if they have backups that must be imported into one version before exporting then importing into a newer version?

1.x already covers 0.x imports by [creating the mobiledoc field](https://github.com/TryGhost/Ghost/blob/master/core/server/data/importer/importers/data/posts.js#L164-L182) if it doesn't exist so I think there are two extra steps required:

1. [ensure image size attributes are correct](https://github.com/TryGhost/Ghost/blob/master/core/server/data/migrations/versions/1.25/1-update-koenig-beta-html.js#L47-L53)
2. re-render every post so the rendered HTML matches the Ghost version (we might already do this? I wasn't sure if [`models.Post.add`](https://github.com/TryGhost/Ghost/blob/master/core/server/data/importer/importers/data/base.js#L306) runs through the normal [`onSaving`](https://github.com/TryGhost/Ghost/blob/master/core/server/models/post.js#L294) hook

Reply:
Yeah i thought about this in the morning again. We should support all 1.0 exports, otherwise there are cases which are annoying.

Reply:
I'll push some more importer tests and logic.